lib/cartopy/crs.py

```diff
@@ -338,43 +339,98 @@ def boundary_distance(xy):
         # Record the positions of all the segment ends
         for i, line_string in enumerate(line_strings):
             first_dist = boundary_distance(line_string.coords[0])
-            thing = _Thing(first_dist, False,
-                           (i, 'first', line_string.coords[0]))
+            thing = _BoundaryPoint(first_dist, False,
```

Comment:
Refactor needed - I'm just going to lift the whole lot out of crs.py. And yes, there will be no mention of "thing" anywhere :wink: .